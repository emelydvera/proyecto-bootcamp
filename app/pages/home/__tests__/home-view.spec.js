const React = require("react");
const View = require("../view");
const { render, screen } = require("@testing-library/react");

describe("Home View", () => {
  let component;
  let i18n = { gettext: (text) => text };
  let originalLocation;

  beforeAll(() => {
    originalLocation = window.location;
    delete window.location;
    window.location = { href: "/" };
  });

  beforeEach(() => {
    component = render(<View i18n={i18n} />);
  });

  it("Should render correctly", () => {
    const { asFragment } = component;
    expect(asFragment()).toMatchSnapshot();
  });
});
