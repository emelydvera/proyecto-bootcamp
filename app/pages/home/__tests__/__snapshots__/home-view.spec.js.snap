// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home View Should render correctly 1`] = `
<DocumentFragment>
  <div
    class="search__container"
  >
    <form
      class="search__form"
    >
      <div
        class="andes-form-control andes-form-control--textfield andes-form-control--default search__form__input"
      >
        <label>
          <div
            class="andes-form-control__control"
          >
            <input
              aria-label="Ingrese producto a buscar"
              class="andes-form-control__field"
              maxlength="120"
              placeholder="Buscar productos, marcas y mÃ¡s..."
              rows="1"
              tabindex="11"
              type="text"
              value=""
            />
          </div>
        </label>
        <div
          class="andes-form-control__bottom"
        >
          <span
            class="andes-form-control__message"
          />
        </div>
      </div>
      <button
        class="andes-button search__form__button andes-button--large andes-button--loud andes-button--disabled"
        disabled=""
        type="submit"
      >
        <span
          class="andes-button__content"
        >
          <span
            class="andes-button__text"
          >
            Buscar
          </span>
        </span>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
